RecoTracker/PixelVertexFinding/plugins/alpaka/vertexFinder.dev.cc

```diff
@@ -0,0 +1,211 @@
+#include <alpaka/alpaka.hpp>
+#include "DataFormats/TrackSoA/interface/alpaka/TrackUtilities.h"
+#include "DataFormats/Vertex/interface/alpaka/ZVertexUtilities.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/traits.h"
+#include "PixelVertexWorkSpaceUtilitiesAlpaka.h"
+#include "PixelVertexWorkSpaceSoADeviceAlpaka.h"
+
+#include "vertexFinder.h"
+#include "vertexFinder.h"
+#include "clusterTracksDBSCAN.h"
+#include "clusterTracksIterative.h"
+#include "clusterTracksByDensity.h"
+#include "fitVertices.h"
+#include "sortByPt2.h"
+#include "splitVertices.h"
+
+#undef PIXVERTEX_DEBUG_PRODUCE
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  namespace vertexFinder {
+    using namespace cms::alpakatools;
+    // reject outlier tracks that contribute more than this to the chi2 of the vertex fit
+    constexpr float maxChi2ForFirstFit = 50.f;
+    constexpr float maxChi2ForFinalFit = 5000.f;
+
+    // split vertices with a chi2/NDoF greater than this
+    constexpr float maxChi2ForSplit = 9.f;
+
+    template <typename TrackerTraits>
+    class loadTracks {
+    public:
+      template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+      ALPAKA_FN_ACC void operator()(const TAcc& acc,
+                                    TrackSoAConstView<TrackerTraits> tracks_view,
+                                    VtxSoAView soa,
+                                    WsSoAView pws,
+                                    float ptMin,
+                                    float ptMax) const {
+        auto const* quality = tracks_view.quality();
+        using helper = TracksUtilities<TrackerTraits>;
+
+        for (auto idx : cms::alpakatools::elements_with_stride(acc, tracks_view.nTracks())) {
+          // TODO: since nHits is not used anywhere else it gives of an unused variable warning. Check!
+          // auto nHits = helper::nHits(tracks_view, idx);
+          // ALPAKA_ASSERT_OFFLOAD(nHits >= 3);
+
+          // initialize soa...
+          soa[idx].idv() = -1;
+
+          if (helper::isTriplet(tracks_view, idx))
+            continue;  // no triplets
+          if (quality[idx] < ::pixelTrack::Quality::highPurity)
+            continue;
+
+          auto pt = tracks_view[idx].pt();
+
+          if (pt < ptMin)
+            continue;
+
+          // clamp pt
+          pt = std::min<float>(pt, ptMax);
+
+          auto& data = pws;
+          auto it = alpaka::atomicAdd(acc, &data.ntrks(), 1u, alpaka::hierarchy::Blocks{});
+          data[it].itrk() = idx;
+          data[it].zt() = helper::zip(tracks_view, idx);
+          data[it].ezt2() = tracks_view[idx].covariance()(14);
+          data[it].ptt2() = pt * pt;
+        }
+      }
+    };
+// #define THREE_KERNELS
+#ifndef THREE_KERNELS
+    class vertexFinderOneKernel {  //FIXME: CUDA do not have the split here
```

Comment:
```suggestion
    class VertexFinderOneKernel {  //FIXME: CUDA do not have the split here
```
Does the FIXME need to be addressed?